django/utils/http.py

```diff
@@ -372,7 +373,7 @@ def _is_safe_url(url, allowed_hosts, require_https=False):
     if not url_info.scheme and url_info.netloc:
         scheme = 'http'
     valid_schemes = ['https'] if require_https else ['http', 'https']
-    return ((not url_info.netloc or url_info.netloc in allowed_hosts) and
+    return ((not url_info.netloc or url_info.netloc in allowed_hosts or '*' in allowed_hosts) and
```

Comment:
There's a very specific, very unlikely problem scenario here, I think? If a typo is given in the allowed hosts, such that `allowed_hosts = ('something*')` (where perhaps the user meant to type `8`, or missed a comma) that would now make everything be considered a safe URL because it's not a tuple but a string, but `x in y` works on strings and the single string now *contains* a `*` but is not *just* `*` ...
Would `or {allowed_hosts} == {'*'}` work in it's place?

Reply:
@kezabelle 
The same problem occurs in code from master:

`django/django/http/request.py:571`
If allowed hosts is string with '*' inside, than all hosts are accepted.

In my opinion better option is:
`or allowed_hosts == '*'`

Is it acceptable?

Reply:
Thanks for checking the usage elsewhere. If the same pattern exists elsewhere (you're right, it does), then its needlessly pedantic of me. It's fine as is.

Reply:
@kezabelle 
So i left this as it was before, added subtest to test method and rebased this PR. This PR is ready for second iteration of QA.